packages/@react-spectrum/s2/src/SelectBoxGroup.tsx

```diff
@@ -0,0 +1,300 @@
+  
+  let {
+    children,
+    onSelectionChange,
+    selectedKeys: controlledSelectedKeys,
+    defaultSelectedKeys,
+    selectionMode = 'single',
+    orientation = 'vertical',
+    numColumns = 2,
+    gutterWidth = 'default',
+    isDisabled = false,
+    isCheckboxSelection = false,
+    name,
+    UNSAFE_style,
+    'aria-label': ariaLabel,
+    'aria-labelledby': ariaLabelledby
+  } = props;
+
+  const domRef = useDOMRef(ref);
+
+  const [selectedKeys, setSelectedKeys] = useControlledState(
+    controlledSelectedKeys,
+    defaultSelectedKeys || new Set(),
+    onSelectionChange
+  );
+
+  const childrenArray = React.Children.toArray(children).filter((x) => x);
+  
+  const disabledKeys = useMemo(() => {
+    const disabled = new Set<string>();
+    
+    childrenArray.forEach((child, index) => {
+      if (React.isValidElement(child)) {
+        const childElement = child as ReactElement<{value?: string, isDisabled?: boolean}>;
+        const childValue = childElement.props?.value || String(index);
+        
+        if (isDisabled || childElement.props?.isDisabled) {
+          disabled.add(childValue);
+        }
+      }
+    });
+    
+    return disabled.size > 0 ? disabled : undefined;
+  }, [isDisabled, childrenArray]);
+  
+  useEffect(() => {
+    if (childrenArray.length > 9) {
+      console.error('Invalid content. SelectBoxGroup cannot have more than 9 children.');
+    }
+  }, [childrenArray.length]);
+
+  const selectBoxContextValue = useMemo(
+    () => {
+      const contextValue = {
+        allowMultiSelect: selectionMode === 'multiple',
+        orientation,
+        isDisabled,
+        isCheckboxSelection,
+        selectedKeys,
+        onSelectionChange: setSelectedKeys
+      };
+      
+      return contextValue;
+    },
+    [selectionMode, orientation, isDisabled, isCheckboxSelection, selectedKeys, setSelectedKeys]
+  );
+
+  return (
+    <div
+      className={containerStyles(null, props.styles)}
+      style={UNSAFE_style}
+      ref={domRef}>
+      
+      <FormIntegration
+        name={name}
+        selectedKeys={selectedKeys}
+        selectionMode={selectionMode} />
+      
+      <ListBox
+        layout="grid"
+        selectionMode={selectionMode}
+        selectedKeys={selectedKeys}
+        onSelectionChange={setSelectedKeys}
+        disabledKeys={disabledKeys}
+        aria-label={ariaLabel || 'Select options'}
+        aria-labelledby={ariaLabelledby}
+        className={gridStyles({gutterWidth, orientation}, props.styles)}
+        style={{
+          gridTemplateColumns: `repeat(${numColumns}, 1fr)`
+        }}>
+        
+        {childrenArray.map((child, index) => {
+          if (!React.isValidElement(child)) {return null;}
+          
+          const childElement = child as ReactElement<{value?: string}>;
+          const childValue = childElement.props?.value || String(index);
+          
+          const getTextValue = (element: ReactElement): string => {
```

Comment:
talked with rob, we'll just have users pass a textValue so make it a prop and you won't have to determine what it is